src/config.rs

```diff
@@ -1,33 +1,140 @@
+
+        #[cfg(debug_assertions)]
+        println!("{:?}", home);
+
         Self {
-            cmd: "ls".into(),
+            cmd: bin.into(),
             numprocs: 1,
-            workingdir: PathBuf::from("/tmp"),
+            umask: 0o027,
+            workingdir: home.clone(),
+            autostart: true,
+            autorestart: Restart::Never,
+            exitcodes: {
+                let mut v = Vec::new();
+                v.push(0);
+                v
+            },
+            startretries: 0,
+            starttime: 0,
+            stopsignal: "SIGTERM".to_string(),
+            stoptime: 0,
+            stdout: home.join(bin.to_string() + ".stdout"), // Would be lovely to replace "bin" by "cmd" from the struct itself
+            stderr: home.join(bin.to_string() + ".stderr"), // Same
+            env: HashMap::new(),
         }
     }
 }
 
-impl std::convert::TryFrom<&Path> for Config {
-    type Error = std::io::Error;
+/// # Errors
+///
+/// Will return `Err` if `file` does not exist or the user does not have
+/// permission to read it or yaml format is not respected.
+pub fn parse(
+    file_name: &Path,
+) -> Result<HashMap<String, HashMap<String, Config>>, serde_yaml::Error> {
+    let file = File::open(&file_name).expect("Unable to open config file");
+    let d: HashMap<String, HashMap<String, Config>> = serde_yaml::from_reader(file)?;
+    Ok(d)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn empty_default_values() {
+        let conf = Config::default();
+        assert_eq!(conf.cmd, "ls");
+        assert_eq!(conf.exitcodes[0], 0);
+        assert_eq!(conf.workingdir, dirs::home_dir().unwrap());
+        assert_eq!(conf.stdout, dirs::home_dir().unwrap().join("ls.stdout"));
+        assert_eq!(conf.stderr, dirs::home_dir().unwrap().join("ls.stderr"));
+    }
+
+    #[test]
+    fn prg_default_values() {
+        let conf = Config {
+            cmd: "nginx".to_string(),
+            ..Config::default()
+        };
+        assert_eq!(conf.cmd, "nginx");
+        assert_eq!(conf.exitcodes[0], 0);
+        assert_eq!(conf.workingdir, dirs::home_dir().unwrap());
+        // assert_ne!(conf.stdout, dirs::home_dir().unwrap().join("ls.stdout"));
+        // assert_ne!(conf.stderr, dirs::home_dir().unwrap().join("ls.stderr"));
+    }
 
-    fn try_from(_file_name: &Path) -> Result<Self, Self::Error> {
-        // TODO: deserialize from yaml file
-        Ok(Self::default())
+    #[test]
+    fn config_files() {
+        assert!(parse(Path::new("tests/config.yaml")).is_ok());
+        assert!(parse(Path::new("tests/empty_section.yaml")).is_err());
+        assert!(parse(Path::new("tests/empty.yaml")).is_err());
+        assert!(parse(Path::new("tests/lot_sections.yaml")).is_ok());
+        assert!(parse(Path::new("tests/no_vec.yaml")).is_err());
+        assert!(parse(Path::new("tests/num_section.yaml")).is_ok());
+        assert!(parse(Path::new("tests/unknow_section.yaml")).is_err());
```

Comment:
Ideally we'd want to check that `parse()` returns correct `Config` files, not just that they error'd or not.